Author: Andreas Beckmann <debian@abeckmann.de>,
 Rico Tzschichholz <ricotz@ubuntu.com>
Description: build a shared library for libXNVCtrl
Bug-Debian: https://bugs.debian.org/666909

--- a/src/Makefile
+++ b/src/Makefile
@@ -273,9 +273,9 @@
 	$(INSTALL) $(INSTALL_BIN_ARGS) $< $(BINDIR)/$(notdir $<)
 
 $(eval $(call DEBUG_INFO_RULES, $(NVIDIA_SETTINGS)))
-$(NVIDIA_SETTINGS).unstripped: $(OBJS) $(LIBXNVCTRL)
+$(NVIDIA_SETTINGS).unstripped: $(OBJS) $(LIBXNVCTRL_SHARED)
 	$(call quiet_cmd,LINK) $(CFLAGS) $(LDFLAGS) $(BIN_LDFLAGS) \
-	    -rdynamic -o $@ $(OBJS) $(LIBXNVCTRL) $(LIBS)
+	    -rdynamic -o $@ $(OBJS) $(LIBXNVCTRL_SHARED) $(LIBS)
 
 $(eval $(call DEBUG_INFO_RULES, $(GTK2LIB)))
 $(GTK2LIB).unstripped: $(GTK2_OBJS) $(XCP_OBJS) $(IMAGE_OBJS) $(VERSION_MK)
@@ -283,7 +283,7 @@
 	    -o $@ \
 	    -Wl,-soname -Wl,$(GTK2LIB_SONAME) \
 	    $(GTK2_OBJS) $(IMAGE_OBJS) $(XCP_OBJS) \
-	    $(LIBXNVCTRL) $(LIBS) $(GTK2_LIBS)
+	    $(LIBXNVCTRL_SHARED) $(LIBS) $(GTK2_LIBS)
 
 ifdef BUILD_GTK3LIB
 $(eval $(call DEBUG_INFO_RULES, $(GTK3LIB)))
@@ -292,7 +292,7 @@
 	    -o $@ \
 	    -Wl,-soname -Wl,$(GTK3LIB_SONAME) \
 	    $(GTK3_OBJS) $(XCP_OBJS) $(IMAGE_OBJS) \
-	    $(LIBXNVCTRL) $(LIBS) $(GTK3_LIBS)
+	    $(LIBXNVCTRL_SHARED) $(LIBS) $(GTK3_LIBS)
 endif
 
 # define the rule to build each object file
@@ -303,7 +303,7 @@
 	rm -rf $(NVIDIA_SETTINGS) *~ \
 		$(OUTPUTDIR)/*.o $(OUTPUTDIR)/*.d \
 		$(GTK2LIB) $(GTK3LIB) $(GTK2LIB_DIR) $(GTK3LIB_DIR) \
-		$(IMAGE_HEADERS) $(LIBXNVCTRL)
+		$(IMAGE_HEADERS) $(LIBXNVCTRL_SHARED)
 
 $(foreach src,$(GTK_SRC), \
     $(eval $(call DEFINE_OBJECT_RULE_WITH_DIR,TARGET,$(src),$(GTK2LIB_DIR))))
--- a/src/libXNVCtrl/Makefile
+++ b/src/libXNVCtrl/Makefile
@@ -51,7 +51,9 @@ LDFLAGS += $(XNVCTRL_LDFLAGS)
 .PHONY: clean
 
 all: $(LIBXNVCTRL)
+all: $(LIBXNVCTRL_SHARED)
 
 clean:
 	rm -rf $(LIBXNVCTRL) *~ \
 		$(OUTPUTDIR)/*.o $(OUTPUTDIR)/*.d
+	rm -f $(LIBXNVCTRL_SHARED) $(LIBXNVCTRL_SHARED).*
--- a/src/libXNVCtrl/xnvctrl.mk
+++ b/src/libXNVCtrl/xnvctrl.mk
@@ -37,7 +37,8 @@ endif
 XNVCTRL_CFLAGS ?=
 
 
-LIBXNVCTRL = $(OUTPUTDIR)/libXNVCtrl.a
+LIBXNVCTRL = libXNVCtrl.a
+LIBXNVCTRL_SHARED = libXNVCtrl.so
 
 LIBXNVCTRL_SRC = $(XNVCTRL_DIR)/NVCtrl.c
 
@@ -47,3 +48,9 @@ $(eval $(call DEFINE_OBJECT_RULE,TARGET,
 
 $(LIBXNVCTRL) : $(LIBXNVCTRL_OBJ)
 	$(call quiet_cmd,AR) ru $@ $(LIBXNVCTRL_OBJ)
+
+$(LIBXNVCTRL_SHARED): $(LIBXNVCTRL_OBJ)
+	$(RM) $@ $@.*
+	$(CC) -shared -fPIC -Wl,-soname=$@.0 -o $@.0.0.0 $(LDFLAGS) $^ -lXext -lX11
+	ln -s $@.0.0.0 $@.0
+	ln -s $@.0 $@
